### PR TITLE
AWS2 Extensions cleanup

### PR DESCRIPTION
**Pull request by oscerd**
[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

File: docs/modules/ROOT/pages/list-of-camel-quarkus-extensions.adoc [hunk 1]
```diff
@@ -241,7 +241,7 @@ Level | Since | Description
 `dozer:name` | Native +
  Stable | 1.0.0-M1 | The dozer component provides the ability to map between Java beans using the Dozer mapping library.
 
-| link:https://camel.apache.org/components/latest/elasticsearch-rest-component.html[Elasticsearch Rest] (camel-quarkus-elasticsearch-rest) +
+| link:https://camel.apache.org/components/latest/elasticsearch-rest-component.html[Elastichsearch Rest] (camel-quarkus-elasticsearch-rest) +
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/list-of-camel-quarkus-extensions.adoc:244]: `Elastichsearch` does not sound right.

**Comment by oscerd** [comment]
It's autogenerated. No idea

**Comment by oscerd** [comment]
I'll rebuild

File: docs/modules/ROOT/pages/list-of-camel-quarkus-extensions.adoc (hunk 1 above)
Comment (oscerd) [reply docs/modules/ROOT/pages/list-of-camel-quarkus-extensions.adoc]: Fixed

File: extensions/aws2-s3/deployment/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/Aws2S3Processor.java [hunk 2]
```diff
@@ -48,17 +46,9 @@ FeatureBuildItem feature() {
         return new FeatureBuildItem(FEATURE);
     }
 
-    @BuildStep
-    ExtensionSslNativeSupportBuildItem activateSslNativeSupport() {
```
Comment (ppalaga) [inline extensions/aws2-s3/deployment/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/Aws2S3Processor.java:52]: Shouldn't we produce ExtensionSslNativeSupportBuildItem in AwsCommonsProcessor if we remove it here? I guess the AWS API client communicates over HTTPS, so SSL is needed.

Comment (oscerd) [reply extensions/aws2-s3/deployment/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/Aws2S3Processor.java:52]: Tests are working

Comment (oscerd) [reply extensions/aws2-s3/deployment/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/Aws2S3Processor.java:52]: Btw I'll check again tomorrow

**Pull request merged**